src/stan/lang/function_signatures.h

```diff
@@ -123,24 +123,43 @@ for (size_t i = 0; i < vector_types.size(); ++i) {
     }
   }
 }
-add_binary("beta_rng");
+for (size_t i = 0; i < primitive_types.size(); ++i) {
+  for (size_t j = 0; j < primitive_types.size(); ++j) {
+    add("beta_rng", expr_type(double_type()), primitive_types[i], primitive_types[j]);
+  }
+ }
+for (size_t i = 0; i < primitive_types.size(); i++) {
+  add("beta_rng", expr_type(vector_type()), primitive_types[i], expr_type(int_type(), 1U));
+  add("beta_rng", expr_type(vector_type()), expr_type(int_type(), 1U), primitive_types[i]);
+ }
+for (size_t i = 1; i < vector_types.size(); ++i) {
+  for (size_t j = 0; j < primitive_types.size(); ++j) {
+    add("beta_rng", expr_type(vector_type()), primitive_types[j], vector_types[i]);
+    add("beta_rng", expr_type(vector_type()), vector_types[i], primitive_types[j]);
+  }
+  add("beta_rng", expr_type(vector_type()), vector_types[i], expr_type(int_type(), 1U));
+  add("beta_rng", expr_type(vector_type()), expr_type(int_type(), 1U), vector_types[i]);
```

Comment:
Must fix:  The vector types include all the primitive types.  So this will add a two integer argument version with a vector return type.  The loops should cover only the required cases.  

There are a few ways to do this.  One would be to define a vector-only container of types, which is what I think you were thinking that `vector_types` was.

Another would be to define a vectorized return type function that would let you just write this all as a big double loop that would test, as in:

```
for (const auto& t : vector_types)
  for (const auto& u : vector_types)
    add("beta_rng", rng_return_type<double>(t, u), t, u);
```

where that `rng_return_type` would return `double` (its template arg so the same thing will work for vectorized `int` forms) if both the arguments were primitives.  I would prefer this latter form as it's much less code everywhere.  Something like:

```
expr_type
rng_return_type(const expr_type& t, const expr_type& u) {
  return is_primitive(t) && is_primitive(u)
    ? expr_type(double_type()) : expr_type(vector_type());
}
```
The `expr_type()` constructors may not be necessary if they're defined as implicit.  This assumes there's somehting like an `is_primitive(t)` test on the types.  If not, that'll have to be written, too.

Reply:
Oh, that is sneaky.  Please don't do that kind of thing.  Code readability has to be priority number one for maintainability.  Better to make your own collection, because the vector types collection might change ordering.